web/impact/impact/tests/test_algolia_api_key_view.py

```diff
@@ -53,10 +81,24 @@ def test_unauthenticated_user_is_denied(self):
                 response_data['detail'] == 'Authentication credentials '
                                            'were not provided.')
 
-    def test_staff_user_gets_empty_filters(self):
+    def test_entrepreneur_that_never_had_a_finalist_role_gets_403(self):
```

Comment:
how is this different from the test at line 38?

Reply:
This one tests an Entrant user i.e. they do not have a RoleGrant to anything, the other tests for an Entrepreneur whose RoleGrants are all for ended programs. I was testing both use cases of this condition in the `DirectoryAccessPermissions` class

https://github.com/masschallenge/impact-api/blob/95f528908b7c4775f517eac2a0dc2a3e20b93802/web/impact/impact/permissions.py#L87

Reply:
Okay, but they're identical. Not sure that we need both.  :)

```
    def test_logged_in_user_with_no_role_grant_gets_403(self):
        user = self._create_entrepreneur()
        with self.settings(
                ALGOLIA_APPLICATION_ID='test',
                ALGOLIA_API_KEY='test'):
            with self.login(email=user.email):
                response = self.client.get(self.url)
                self.assertTrue(response.status_code, 403)

def test_entrepreneur_that_never_had_a_finalist_role_gets_403(self):
        user = self._create_entrepreneur()
        with self.settings(
                ALGOLIA_APPLICATION_ID='test',
                ALGOLIA_API_KEY='test'):
            with self.login(email=user.email):
                response = self.client.get(self.url)
                self.assertTrue(response.status_code, 403)
```

Reply:
ah, those two..I was comparing the wrong two. I'll get rid of the first. Thanks.